lib/src/realm_class.dart

```diff
@@ -474,3 +464,74 @@ class RealmLogLevel {
   /// Same as [Level.OFF];
   static const off = Level.OFF;
 }
+
+/// @nodoc
+class RealmMetadata {
+  final Map<Type, RealmObjectMetadata> _typeMap = <Type, RealmObjectMetadata>{};
+  final Map<String, RealmObjectMetadata> _stringMap = <String, RealmObjectMetadata>{};
+
+  RealmMetadata._(Iterable<RealmObjectMetadata> objects) {
+    for (final meta in objects) {
+      if (meta.type != RealmObject) {
+        _typeMap[meta.type] = meta;
+      } else {
+        _stringMap[meta.name] = meta;
+      }
+    }
+  }
+
+  RealmObjectMetadata getByType(Type type) {
+    final metadata = _typeMap[type];
+    if (metadata == null) {
+      throw RealmError("Object type $type not configured in the current Realm's schema. Add type $type to your config before opening the Realm");
+    }
+
+    return metadata;
+  }
+
+  RealmObjectMetadata getByName(String type) {
+    var metadata = _stringMap[type];
+    if (metadata == null) {
+      metadata = _typeMap.values.firstWhereOrNull((v) => v.name == type);
+      if (metadata == null) {
+        throw RealmError("Object type $type not configured in the current Realm's schema. Add type $type to your config before opening the Realm");
+      }
+
+      _stringMap[type] = metadata;
+    }
+
+    return metadata;
+  }
+}
+
+/// Exposes a set of dynamic methods on the Realm object. These don't use strongly typed
+/// classes and instead lookup objects by string name.
+///
+/// {@category Realm}
+class DynamicRealm {
```

Comment:
Should this class implement Realm class. Otherwise how are we going to keep up the interfaces in sync

Reply:
No, it provides a completely different API than the Realm class.

Reply:
I guess my question should be why not merge these methods in Realm class or have these as extension methods. Why do we need a DynamicRealm concept?